runtime/nvqir/CircuitSimulator.h

```diff
@@ -807,7 +807,8 @@ class CircuitSimulatorBase : public CircuitSimulator {
   std::size_t allocateQubit() override {
     // Get a new qubit index
     auto newIdx = tracker.getNextIndex();
-
+    const bool inTraceMode =
+        (executionContext && executionContext->name == "tracer");
```

Comment:
Looks like you could make this a small `static` function in this file.

Reply:
Indeed, fixed in https://github.com/NVIDIA/cuda-quantum/pull/1607/commits/96b1112502e95b34cec0c9271b05ffe4035667cf